src/core/communication.cpp

```diff
@@ -257,32 +259,27 @@ void mpi_init(int *argc, char ***argv) {
   }
 #endif
 
-  MPI_Init(argc, argv);
+  mpi_env =
+      Utils::make_unique<boost::mpi::environment>(*argc, *argv);
```

Comment:
On mac os, i get a segfault on exactly this line:

thread #2: tid = 0x1dfaf, 0x00000001023baeeb libmpi.12.dylib`MPI_Comm_set_errhandler + 219, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
  * frame #0: 0x00000001023baeeb libmpi.12.dylib`MPI_Comm_set_errhandler + 219
    frame #1: 0x00000001024ca494 libboost_mpi-mt.dylib`boost::mpi::environment::environment(int&, char**&, bool) + 84
    frame #2: 0x00000001026d4deb libEspressoCore.4.dylib`mpi_init(argc=0x0000000000000000, argv=0x0000000000000000) + 651 at communication.cpp:268
    frame #3: 0x00000001019cabec _init.so`::init_init() + 1948 at _init.cpp:1090
    frame #4: 0x00000001001662ec Python`_PyImport_LoadDynamicModule + 140
    frame #5: 0x0000000100164cef Python`import_submodule + 271
    frame #6: 0x0000000100164778 Python`load_next + 280
    frame #7: 0x000000010016380e Python`PyImport_ImportModuleLevel + 1230
    frame #8: 0x000000010013c7c7 Python`builtin___import__ + 135
    frame #9: 0x000000010009e203 Python`PyObject_Call + 99
    frame #10: 0x0000000100147429 Python`PyEval_EvalFrameEx + 22425
    frame #11: 0x0000000100141947 Python`PyEval_EvalCodeEx + 2119
    frame #12: 0x00000001001410f6 Python`PyEval_EvalCode + 54
    frame #13: 0x00000001001621a1 Python`PyImport_ExecCodeModuleEx + 241
    frame #14: 0x0000000100165273 Python`load_source_module + 1059
    frame #15: 0x00000001001655ef Python`load_package + 303
    frame #16: 0x0000000100164cef Python`import_submodule + 271
    frame #17: 0x0000000100164778 Python`load_next + 280
    frame #18: 0x00000001001637b2 Python`PyImport_ImportModuleLevel + 1138
    frame #19: 0x000000010013c7c7 Python`builtin___import__ + 135
    frame #20: 0x000000010009e203 Python`PyObject_Call + 99
    frame #21: 0x0000000100147429 Python`PyEval_EvalFrameEx + 22425
    frame #22: 0x0000000100141947 Python`PyEval_EvalCodeEx + 2119
    frame #23: 0x00000001001410f6 Python`PyEval_EvalCode + 54
    frame #24: 0x0000000100170824 Python`PyRun_FileExFlags + 164
    frame #25: 0x000000010017035e Python`PyRun_SimpleFileExFlags + 702
    frame #26: 0x0000000100186b1d Python`Py_Main + 2925
    frame #27: 0x00007fffbbe11235 libdyld.dylib`start + 1
    frame #28: 0x00007fffbbe11235 libdyld.dylib`start + 1